ibis/expr/window.py

```diff
@@ -71,7 +78,16 @@ def __init__(
                 x = ops.SortKey(x).to_expr()
             self._order_by.append(x)
 
-        self.preceding = _sequence_to_tuple(preceding)
+        if isinstance(preceding, dict):
```

Comment:
What do you think about making the API a top level function `ibis.rows_with_max_lookback(3, '5D')` and having that produce a special `namedtuple` named `RowsWithMaxLookback`? I think a `dict` is a bit too flexible and exposes the details of the underlying data structure as opposed to exposing a function whose details can be mostly ignored. A namedtuple would actually restrict the fields of the thing produced by the function, which would allow us to assume existing fields in that tuple. That way we don't have to check for existing keys: we know that if we received an object of type `RowsWithMaxLookback` then it has the fields we care about.